sdk/modelsrepository/Azure.Iot.ModelsRepository/src/Fetchers/LocalModelFetcher.cs

```diff
@@ -27,35 +30,48 @@ public Task<FetchResult> FetchAsync(string dtmi, Uri repositoryUri, Cancellation
 
         public FetchResult Fetch(string dtmi, Uri repositoryUri, CancellationToken cancellationToken = default)
         {
-            var work = new Queue<string>();
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope("LocalModelFetcher.Fetch");
+            scope.Start();
 
-            if (_tryExpanded)
+            try
             {
-                work.Enqueue(GetPath(dtmi, repositoryUri, true));
-            }
+                var work = new Queue<string>();
 
-            work.Enqueue(GetPath(dtmi, repositoryUri, false));
+                if (_tryExpanded)
+                {
+                    work.Enqueue(GetPath(dtmi, repositoryUri, true));
+                }
 
-            string fnfError = string.Empty;
-            while (work.Count != 0 && !cancellationToken.IsCancellationRequested)
-            {
-                string tryContentPath = work.Dequeue();
-                ResolverEventSource.Shared.FetchingModelContent(tryContentPath);
+                work.Enqueue(GetPath(dtmi, repositoryUri, false));
 
-                if (File.Exists(tryContentPath))
+                string fnfError = string.Empty;
+                while (work.Count != 0 && !cancellationToken.IsCancellationRequested)
```

Comment:
Should remove the check for IsCancellationRequested, no?

It'll throw on line 50 if it is, and that code won't get hit if we check for it here.

Reply:
:D I forgot